src/OpenApi/src/Services/OpenApiDocumentService.cs

```diff
@@ -154,4 +160,66 @@ private static OpenApiOperation GetOperation(ApiDescription description, HashSet
         // allows us to group endpoints by the "resource" concept (e.g. users, todos, etc.)
         return [new OpenApiTag { Name = description.ActionDescriptor.RouteValues["controller"] }];
     }
+
+    private static OpenApiResponses GetResponses(ApiDescription description)
+    {
+        // OpenAPI requires that each operation have a response, usually a successful one.
+        // if there are no response types defined, we assume a successful 200 OK response
+        // with no content by default.
+        if (description.SupportedResponseTypes.Count == 0)
+        {
+            return new OpenApiResponses
+            {
+                ["200"] = GetResponse(description, StatusCodes.Status200OK, _defaultApiResponseType)
+            };
+        }
+
+        var responses = new OpenApiResponses();
+        foreach (var responseType in description.SupportedResponseTypes)
+        {
+            // The "default" response type is a special case in OpenAPI used to describe
+            // the response for all HTTP status codes that are not explicitly defined
+            // for a given operation. This is typically used to describe catch-all scenarios
+            // like error responses.
+            var statusCode = responseType.IsDefaultResponse
+                ? OpenApiConstants.DefaultResponseStatusCode
+                : responseType.StatusCode.ToString(CultureInfo.InvariantCulture);
+            responses.Add(statusCode, GetResponse(description, responseType.StatusCode, responseType));
```

Comment:
Personally, I find it a bit confusing that there are two status codes in this line and they're not the same.   What's the value of `responseType.StatusCode` when `statusCode` is `"default"`?

Reply:
Per our offline conversation, I've updated certain uses of the `statusCode` here to `responseKey` to more accurately describe what they are doing.